doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+.. _functional_in_depth:
+
+#####################
+Functional
+#####################
+
+A *functional* is an operator ``f`` that maps from some vector space ``X`` to the field of scalars ``F`` associated with the vector space:
+
+.. math::
+   
+   f : X \to F.
+
+In the ODL solver package, functionals are implemented in the `Functional` class, a subclass to `Operator`.
+
+From a mathematical presepective, the above is a valide definition of a functional.
+However, since the purpose of these functionals are primarely to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space ``X`` is a Hilbert space.
```

Comment:
As @kohr-h says. This assumption was made with convex conjugate functional in mind. If you do not assume Hilbert spaces, the dual variable is suppose to live in the normed dual space. So what domain would you return for these functionals?

I think most of the other math still holds, but it should be double checked... for example convex conjugates of convex conjugates would like in the second dual space, so if the space is not reflexive we will be in trouble.

What do you say @adler-j ?